### PR TITLE
WIP DEVX-5712 Feature/selective archive broadcast

### PR DESCRIPTION
**Pull request by moficodes**
#### What is this PR doing?
Adds selective streams to archive and broadcast.

#### What are the relevant tickets?
https://jira.vonage.com/browse/DEVX-5712

### CODE REVIEW

File: lib/opentok.js [hunk 1]
```diff
@@ -327,6 +327,55 @@ OpenTok = function (apiKey, apiSecret, env) {
...
+  *
+  * <ul>
+  *
+  *   <li>
+  *      <code>error</code> &mdash; An error object (if the call to the method fails).
+  *   </li>
+  *
+  * </ul>
+  *
+  * @method #patchArchive
```
Comment (jeffswartz) [inline lib/opentok.js:374]: This does not match the proposal: https://github.com/opentok/api-review/pull/232/. The proposal has two methods: `OpenTok.addArchiveStream()` and `OpenTok.removeArchiveStream()`.

File: lib/opentok.js [hunk 2]
```diff
@@ -515,13 +569,72 @@ OpenTok = function (apiKey, apiSecret, env) {
...
+  *
+  *   <li>
+  *     <code>hasAudio</code> &mdash; Archive the video stream. Default value is true.
+  *   </li>
+  *
+  * </ul>
+  * @method #patchBroadcast
+  * @memberof OpenTok
+  */
+  this.patchBroadcast = function patchBroadcast(broadcastId, options, callback) {
```
Comment (jeffswartz) [inline lib/opentok.js:617]: This does not match the proposal: https://github.com/opentok/api-review/pull/232/. The proposal has two methods: `OpenTok.addBroadcastStream()` and `OpenTok.removeBroadcastStream()`.

**Review by jeffswartz — CHANGES_REQUESTED** [review]

**Review by jeffswartz — CHANGES_REQUESTED** [review]
There are merge conflicts.

**Review by jeffswartz — CHANGES_REQUESTED** [review]
This PR seems to have a lot of changes unrelated to the selective stream archive/broadcast API enhancement. Please add those changes to a separate PR.

**Comment by moficodes** [comment]
#254 closing since that covers this.
WIll make separate pr with lining fixes.

**Pull request closed without merging**